yt_dlp/extractor/manyvids.py

```diff
@@ -1,161 +1,64 @@
-                    'X-Requested-With': 'XMLHttpRequest'
-                })
-
-        formats = []
-        for v_url, fmt in video_urls_and_ids:
-            v_url = url_or_none(v_url)
-            if not v_url:
-                continue
-            if determine_ext(v_url) == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    v_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls'))
-            else:
-                formats.append({
-                    'url': v_url,
-                    'format_id': fmt,
-                })
-
-        self._remove_duplicate_formats(formats)
-
-        for f in formats:
-            if f.get('height') is None:
-                f['height'] = int_or_none(
-                    self._search_regex(r'_(\d{2,3}[02468])_', f['url'], 'video height', default=None))
-            if '/preview/' in f['url']:
-                f['format_id'] = '_'.join(filter(None, (f.get('format_id'), 'preview')))
-                f['preference'] = -10
-            if 'transcoded' in f['format_id']:
-                f['preference'] = f.get('preference', -1) - 1
-
-        def get_likes():
-            likes = self._search_regex(
-                r'''(<a\b[^>]*\bdata-id\s*=\s*(['"])%s\2[^>]*>)''' % (video_id, ),
-                webpage, 'likes', default='')
-            likes = extract_attributes(likes)
-            return int_or_none(likes.get('data-likes'))
-
-        def get_views():
-            return str_to_int(self._html_search_regex(
-                r'''(?s)<span\b[^>]*\bclass\s*=["']views-wrapper\b[^>]+>.+?<span\b[^>]+>\s*(\d[\d,.]*)\s*</span>''',
-                webpage, 'view count', default=None))
+        title = info["title"]
+        if title and is_free is False:
+            title += " (Preview)"
 
+        if is_free:
+            download_url = download["filepath"]
+        else:
+            download_url = download.get("teaser", {}).get("filepath")
+            if download_url is None:
+                raise ValueError('this video has no preview')
+                # They are rare, but they exist. You can find such videos using the following link:
+                # https://www.manyvids.com/Vids/?content_type=1,2,3&other=blocked&search_type=0&sort=13&page=1
         return {
             'id': video_id,
+            'url': download_url,
             'title': title,
-            'formats': formats,
-            'description': txt_or_none(info.get('data-meta-description')),
-            'uploader': txt_or_none(info.get('data-meta-author')),
-            'thumbnail': (
-                url_or_none(info.get('data-meta-image'))
-                or url_or_none(player.get('data-video-screenshot'))),
-            'view_count': get_views(),
-            'like_count': get_likes(),
+            'description': info.get("description") or None,
+            'uploader': info["model"]["displayName"],
+            'like_count': int(info["likes"]),
```

Comment:
All metadata extraction should be non-fatal. Use `traverse_obj()`.

Can we no longer extract `thumbnail` or `view_count`?

Single quotes all over